### PR TITLE
add course deletion

### PR DESCRIPTION
**Pull request by inlanger**

### CODE REVIEW

File: lms/djangoapps/course_api/views.py [hunk 1]
```diff
@@ -206,3 +213,43 @@ def get_queryset(self):
...
+    def get(self, request, *args, **kwargs):
+
+        course_key_string = self.kwargs['course_key_string']
+        try:
+            course_key = CourseKey.from_string(course_key_string)
+        except InvalidKeyError:
+            raise ValidationError("Invalid course_key: '{}'.".format(course_key_string))
+
+        if not modulestore().get_course(course_key):
+            raise ValidationError("Course with '{}' key not found.".format(course_key_string))
```
Comment (aniav) [inline lms/djangoapps/course_api/views.py]: I am not sure this should return `400`. RFC for delete says it should be 204 in case of no entity returned.

**Review by aniav — CHANGES_REQUESTED** [review]

File: lms/djangoapps/course_api/views.py [hunk 2]
```diff
@@ -206,3 +213,43 @@ def get_queryset(self):
...
             filter_=form.cleaned_data['filter_'],
         )
+
+
+@view_auth_classes(is_authenticated=False)
+class CourseDeletionView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Request details for a course
```
Comment (CrowbarKZ) [inline lms/djangoapps/course_api/views.py]: shouldn't it be `Delete a course` ?

**Review by CrowbarKZ — APPROVED** [review]

**Comment by inlanger** [comment]
@aniav Could you check changes for this PR, please?

**Review by aniav — APPROVED** [review]

**Comment by aniav** [comment]
Oh damn, sorry, I didn't know I was blocking you! :)

**Pull request merged**